src/oneD/StFlow.cpp

```diff
@@ -30,9 +30,14 @@ StFlow::StFlow(IdealGasPhase* ph, size_t nsp, size_t points) :
     m_zfixed(Undef),
     m_tfixed(Undef)
 {
+    if (ph->type() == "IdealGas") {
+        m_thermo = (IdealGasPhase*)ph;
```

Comment:
We try to avoid the unchecked C-style casts:
```suggestion
        m_thermo = static_cast<IdealGasPhase*>(ph);
```